test/actionWithdrawLiquidity.test.js

```diff
@@ -265,14 +532,186 @@ describe("ActionWithdrawLiquidity.sol test", function () {
 
     await addFundingTx.transactionResponse.wait();
 
-    const liquidityPoolTokenBalance = await fixedProductMarketMaker.balanceOf(
+    const addFundingTx_2 = await cpk.execTransactions(
```

Comment:
For all the repetitive logic, we should create helpers functions which externalise that logic into separate files, so that in the core ActionWithdrawLiquidity.test file we minmize logic needed to set the test up and focus more on the logic related to the actual action.